### PR TITLE
ejb-in-war quickstart - initial commit

### PR DESCRIPTION
**Pull request by npfr**
Created quickstart 21: WAR deployment of EJBs – EE6 allows for EJBs to ship in a WAR vs. an EAR.

### CODE REVIEW

**Comment by sgilda** [comment]
I've reviewed this for comments and clarity and it looks great! The only point of confusion for me was the button text "Add". My first reaction was to look at the code to see what the name was added to, because in my mind, when I see "Add", I think of adding to a list or a database. You used "Add" because it's adding the name to the response, correct? Button text with "Greet" or "Respond" might more descriptive, but I'm probably being too picky. I'll defer to Pete on this.

**Comment by npfr** [comment]
Thanks for the review. You are not being too picky, it's a copy and paste error from the quickstart I based it on. Don't know how many times I've clicked that button, without reading it!

I'll fix it shortly.

**Comment by sgilda** [comment]
Forgot to mention I successfully deployed and tested it.

**Comment by npfr** [comment]
@sgilda I've made the change you suggested. Could you re-review please?

**Comment by sgilda** [comment]
Thanks Paul. I will.

**Comment by sgilda** [comment]
Thanks Paul. This looks better. Will merge this after SME review.

File: ejb-in-war/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+5. The managed bean is annotated as @SessionScoped, so the same managed bean instance is used for the entire session. This ensures that the message is available when the page reloads and is
+displayed to the user.
+
+The example can be deployed using Maven from the command line or from Eclipse using JBoss Tools.
+
+To set up Maven or JBoss Tools in Eclipse, refer to the _Getting Started Guide_.
+
+To deploy to JBoss AS 7, start JBoss AS 7, and type `mvn package jboss-as:deploy`.
+The application is deployed to <http://localhost:8080/jboss-as-ejbinwar>.
\ No newline at end of file
```
Comment (pmuir) [inline ejb-in-war/README.md]: Just for consistency, could we align this readme to the others used. For example https://github.com/jbossas/quickstart/blob/master/kitchensink/readme.md

File: ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+import org.jboss.as.quickstarts.ejbinwar.ejb.GreeterEJB;
+import org.jboss.as.quickstarts.ejbinwar.ejb.GreeterEJBImpl;
+
+import javax.ejb.EJB;
+import javax.enterprise.context.SessionScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import java.io.Serializable;
```
Comment (pmuir) [inline ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java]: A few of these imports are unused, do you want to clean them up?

File: ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	 */
+	@EJB
+	private GreeterEJB greeterEJB;
+
+	/*
+	 * Stores the response from the call to greeterEJB.sayHello(...)
+	 */
+	private String message;
+
+	/*
```
Comment (pmuir) [inline ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java]: Probably all these should be javadoc, not just comments.

File: ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2011,
+ * @author JBoss, by Red Hat.
+ */
+
+package org.jboss.as.quickstarts.ejbinwar.jsf;
```
Comment (pmuir) [inline ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/jsf/Greeter.java]: I think this package is possibly misleading. We've generally used "controller" else where (to refer to the c from MVC).

File: ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/ejb/GreeterEJBImpl.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+package org.jboss.as.quickstarts.ejbinwar.ejb;
+
+import javax.ejb.Stateful;
+
+/**
+ * @author paul.robinson@redhat.com, 2011-12-21
+ */
+@Stateful
+public class GreeterEJBImpl implements GreeterEJB {
```
Comment (pmuir) [inline ejb-in-war/src/main/java/org/jboss/as/quickstarts/ejbinwar/ejb/GreeterEJBImpl.java]: As we are showing off the new features of EJB here, what about using a no interface view ejb?

**Comment by pmuir** [comment]
A few minor comments, otherwise this looks great, thanks PauL!

**Comment by npfr** [comment]
Thanks Pete, I'll get onto those changes now. 

I'd hold off reviewing quickstart22 for now as it was based on this quickstart, so also needs the above fixes. I'll comment on that quickstart when I am done there

**Comment by npfr** [comment]
Pete,

I've made the changes you suggested. Thanks for the feedback, looks much better now.

**Comment by npfr** [comment]
Pete,

I've also made the changes to quickstart 22.

**Comment by pmuir** [comment]
Thanks Paul, I squashed all your commits to one, and merged it. I'll take a look at 22 in just a minute.

**Pull request closed without merging**